diff --git a/vcc/Headers/Vcc3Prelude.bpl b/vcc/Headers/Vcc3Prelude.bpl
index d34dedd..319834a 100644
--- a/vcc/Headers/Vcc3Prelude.bpl
+++ b/vcc/Headers/Vcc3Prelude.bpl
@@ -31,7 +31,7 @@ type $token;
 
 type $state;
 
-type $object = [$field][$ptr]int;
+type $object = [$ptr,$field]int;
 type $owner = [$ptr]$ptr;
 type $closed = [$ptr]bool;
 type $timestamps = [$ptr]int;
@@ -563,7 +563,7 @@ function {:inline true} $root(s:$state, p:$ptr) : $ptr
   { $roots(s)[p] }
 
 function {:inline true} $rd_inv(s:$state, f:$field, p:$ptr) : int { $rd(s,p,f) }
-function {:inline true} $rd(s:$state, p:$ptr, f:$field) : int { $heap(s)[f][p] }
+function {:inline true} $rd(s:$state, p:$ptr, f:$field) : int { $heap(s)[p,f] }
 
 function {:inline true} $rd_spec_ptr(s:$state, f:$field, p:$ptr, t:$ctype) : $ptr
   { $spec_ptr_cast($int_to_ptr($rd(s, p, f)), t) }
@@ -840,7 +840,7 @@ axiom (forall S:$state, p:$ptr, f:$field, t:$ctype ::
 // ----------------------------------------------------------------------------
 
 function $update(h:$object, r:$ptr, f:$field, v:int) : $object
-  { h[ f := h[f][ r := v ] ] }
+  { h[ r,f := v ] }
 
 function {:inline true} $specials_eq(S0:$state, S:$state) : bool
   { 
@@ -864,6 +864,7 @@ procedure $write_int(f:$field, p:$ptr, v:int);
   ensures $heap($s) == $update($heap(old($s)), p, f, v);
   ensures $timestamp_post_strict(old($s), $s);
 
+/*
 procedure $write_int_local(h0:[$ptr]int, f:$field, p:$ptr, v:int) returns(h:[$ptr]int);
   modifies $s;
   ensures $specials_eq(old($s), $s);
@@ -872,6 +873,7 @@ procedure $write_int_local(h0:[$ptr]int, f:$field, p:$ptr, v:int) returns(h:[$pt
   ensures h == h0[p := v];
   ensures old($heap($s))[f] == h0;
   ensures $heap($s)[f] == h;
+*/
 
 function {:inline true} $rd_local(h:[$ptr]int, f:$field, p:$ptr) : int
   { h[p] }
